### ISSUE TITLE
Panic when restoring etcd from backup using "all" or "kubernetesVersion" option

### ISSUE THREAD

**Issue by izaac** [rancher/rancher#25609]
**What kind of request is this (question/bug/enhancement/feature request):**

**Steps to reproduce (least amount of steps as possible):**

- Install Rancher 2.4 commit id: 776b0e457
- Create a Linode cluster with S3 backups (used 1 etcd, 1 controlplane, 3 workers) Kubernetes version used for install v1.16 from cluster dropdown.
- Once deployed make sure a backup is created in the S3 bucket 
- Upgrade Kubernetes in this cluster to v1.17
- Create an Snapshot
- Once the Kubernetes cluster is Available again restore from the first snapshot using the UI. It will work.
- From the View API use `restoreFromEtcdBackup` action
- In this action set the value of the `restoreRkeConfig` key to `all`
- Select one of the backups for the other key.
- Click Send Request

**Result:**

Error in the View API UI when adding values: all or kubernetesVersion 

<img width="820" alt="Screen Shot 2020-02-26 at 5 49 32 PM" src="https://user-images.githubusercontent.com/27659/75402951-769ba400-58c3-11ea-87b9-2e53a9f860d7.png">

Rancher logs show:

```bash
2020/02/27 00:57:40 [ERROR] Panic serving api request:
goroutine 44919 [running]:
runtime/debug.Stack(0xc003686ca0, 0x35ee160, 0x6af2560)
	/usr/local/go/src/runtime/debug/stack.go:24 +0x9d
github.com/rancher/rancher/vendor/github.com/rancher/norman/api.(*Server).ServeHTTP.func1(0x45fbe80, 0xc004a61ba0)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/api/server.go:175 +0x7a
panic(0x35ee160, 0x6af2560)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/rancher/rancher/pkg/api/customization/cluster.ActionHandler.RestoreFromEtcdBackupHandler(0x7f519628cbe8, 0xc000e2a000, 0x467b840, 0xc000d239b0, 0x7f519628cc08, 0xc000e2a000, 0x464cd40, 0xc00117ca80, 0xc000937b90, 0x467bfc0, ...)
	/go/src/github.com/rancher/rancher/pkg/api/customization/cluster/actions_etcd.go:117 +0x688
github.com/rancher/rancher/pkg/api/customization/cluster.ActionHandler.ClusterActionHandler(0x7f519628cbe8, 0xc000e2a000, 0x467b840, 0xc000d239b0, 0x7f519628cc08, 0xc000e2a000, 0x464cd40, 0xc00117ca80, 0xc000937b90, 0x467bfc0, ...)
	/go/src/github.com/rancher/rancher/pkg/api/customization/cluster/actionhandler.go:89 +0x1138
github.com/rancher/rancher/vendor/github.com/rancher/norman/api.handleAction(0xc003173fe0, 0xc004a8e6c0, 0x0, 0x0)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/api/server.go:263 +0x62
github.com/rancher/rancher/vendor/github.com/rancher/norman/api.(*Server).handle(0xc0002fa000, 0x45fbe80, 0xc004a61ba0, 0xc004ff0e00, 0xc00517f310, 0xdc6d4800, 0x143906f5d516c53)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/api/server.go:251 +0xd2
github.com/rancher/rancher/vendor/github.com/rancher/norman/api.(*Server).ServeHTTP(0xc0002fa000, 0x45fbe80, 0xc004a61ba0, 0xc004ff0e00)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/api/server.go:180 +0x99
github.com/rancher/rancher/vendor/github.com/rancher/steve/pkg/responsewriter.ContentTypeOptions.func1(0x45fbe80, 0xc004a61ba0, 0xc004ff0e00)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/steve/pkg/responsewriter/cache.go:47 +0x131
net/http.HandlerFunc.ServeHTTP(0xc003173f60, 0x45fbe80, 0xc004a61ba0, 0xc004ff0e00)
	/usr/local/go/src/net/http/server.go:2007 +0x44
github.com/rancher/rancher/vendor/github.com/gorilla/mux.(*Router).ServeHTTP(0xc00117d8c0, 0x45fbe80, 0xc004a61ba0, 0xc004ff0c00)
	/go/src/github.com/rancher/rancher/vendor/github.com/gorilla/mux/mux.go:212 +0xe2
github.com/rancher/rancher/pkg/auth/requests.authHeaderHandler.ServeHTTP(0x45c3400, 0xc0013c2da0, 0x4586e40, 0xc00117d8c0, 0x45d6b40, 0xc000937b90, 0x45c33c0, 0xc0012efc20, 0x45fbe80, 0xc004a61ba0, ...)
	/go/src/github.com/rancher/rancher/pkg/auth/requests/filter.go:113 +0x6f8
github.com/rancher/rancher/pkg/rbac.NewAccessControlHandler.func1(0x45fbe80, 0xc004a61ba0, 0xc004ff0b00, 0x4585fa0, 0xc000d77c00)
	/go/src/github.com/rancher/rancher/pkg/rbac/user_based.go:24 +0x17a
github.com/rancher/rancher/vendor/github.com/rancher/steve/pkg/auth.Middleware.Wrap.func1(0x45fbe80, 0xc004a61ba0, 0xc004ff0b00)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/steve/pkg/auth/filter.go:36 +0x55
net/http.HandlerFunc.ServeHTTP(0xc0013c2f60, 0x45fbe80, 0xc004a61ba0, 0xc004ff0b00)
	/usr/local/go/src/net/http/server.go:2007 +0x44
github.com/rancher/rancher/pkg/websocket.websocketHandler.ServeHTTP(0x458dd00, 0xc0013c2f60, 0x45fbe80, 0xc004a61ba0, 0xc004ff0b00)
	/go/src/github.com/rancher/rancher/pkg/websocket/handler.go:35 +0xa0
github.com/rancher/rancher/pkg/audit.auditHandler.ServeHTTP(0x45861e0, 0xc0004f1510, 0x0, 0x45fbe80, 0xc004a61ba0, 0xc004ff0b00)
	/go/src/github.com/rancher/rancher/pkg/audit/filter.go:31 +0x623
github.com/rancher/rancher/vendor/github.com/rancher/steve/pkg/responsewriter.ContentType.func1(0x45ea300, 0xc005d5dda0, 0xc004ff0b00)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/steve/pkg/responsewriter/content.go:33 +0x93
net/http.HandlerFunc.ServeHTTP(0xc0013e08c0, 0x45ea300, 0xc005d5dda0, 0xc004ff0b00)
	/usr/local/go/src/net/http/server.go:2007 +0x44
github.com/rancher/rancher/vendor/github.com/rancher/steve/pkg/responsewriter.Gzip.func1(0x45fac80, 0xc00851a088, 0xc004ff0b00)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/steve/pkg/responsewriter/gzip.go:60 +0x1f5
net/http.HandlerFunc.ServeHTTP(0xc0013e08e0, 0x45fac80, 0xc00851a088, 0xc004ff0b00)
	/usr/local/go/src/net/http/server.go:2007 +0x44
github.com/rancher/rancher/vendor/github.com/rancher/steve/pkg/responsewriter.ContentTypeOptions.func1(0x45fac80, 0xc00851a088, 0xc004ff0b00)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/steve/pkg/responsewriter/cache.go:47 +0x131
net/http.HandlerFunc.ServeHTTP(0xc003173e80, 0x45fac80, 0xc00851a088, 0xc004ff0b00)
	/usr/local/go/src/net/http/server.go:2007 +0x44
github.com/rancher/rancher/vendor/github.com/gorilla/mux.(*Router).ServeHTTP(0xc00117d800, 0x45fac80, 0xc00851a088, 0xc004ff0500)
	/go/src/github.com/rancher/rancher/vendor/github.com/gorilla/mux/mux.go:212 +0xe2
github.com/rancher/rancher/vendor/github.com/rancher/dynamiclistener/server.wrapHandler.func1(0x45fac80, 0xc00851a088, 0xc004ff0500)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/dynamiclistener/server/server.go:187 +0x95
net/http.HandlerFunc.ServeHTTP(0xc001708540, 0x45fac80, 0xc00851a088, 0xc004ff0500)
	/usr/local/go/src/net/http/server.go:2007 +0x44
github.com/rancher/rancher/vendor/github.com/rancher/dynamiclistener.HTTPRedirect.func1(0x45fac80, 0xc00851a088, 0xc004ff0500)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/dynamiclistener/redirect.go:20 +0x221
net/http.HandlerFunc.ServeHTTP(0xc0016c4440, 0x45fac80, 0xc00851a088, 0xc004ff0500)
	/usr/local/go/src/net/http/server.go:2007 +0x44
net/http.serverHandler.ServeHTTP(0xc000b8fc00, 0x45fac80, 0xc00851a088, 0xc004ff0500)
	/usr/local/go/src/net/http/server.go:2802 +0xa4
net/http.initNPNRequest.ServeHTTP(0x461d000, 0xc003f2e9c0, 0xc004dc7880, 0xc000b8fc00, 0x45fac80, 0xc00851a088, 0xc004ff0500)
	/usr/local/go/src/net/http/server.go:3365 +0x8d
net/http.(*http2serverConn).runHandler(0xc001d47080, 0xc00851a088, 0xc004ff0500, 0xc003173d20)
	/usr/local/go/src/net/http/h2_bundle.go:5713 +0x9f
created by net/http.(*http2serverConn).processHeaders
	/usr/local/go/src/net/http/h2_bundle.go:5447 +0x4eb
```

**Other details that may be helpful:**

- S3 backups were enabled during cluster creation.
- Linode was used for provision the cluster.

**Environment information**
- Rancher version: `2.4 master-head (02/26/2020)` commit id: 776b0e457
- Installation option (single install/HA): single

**Comment by izaac** [comment]
Checked this with the Rancher version `v2.4.0-rc1 (ca71db485)` Using the new snapshots UI options.

- Failed selecting `Kubernetes Version`
- Failed selecting `ETCD and Kubernetes Version`

**Comment by izaac** [comment]
This is fixed in Rancher version `2.4 master-head (02/27/2020)` commit id: 9adb67c4f 

Using S3 storage configuration:

- Worked with `Restore ETCD` ✅ 
- Worked with `Kubernetes Version` ✅ 
- Worked with `ETCD and Kubernetes Version` ✅
